### PR TITLE
Add & populate new fields on ciivcrm_mailing - start_date, end_date, status (allows deleting of completed civicrm_mailing_job records)

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Builds on https://github.com/civicrm/civicrm-core/pull/27559 but adds the upgrade script - I think the other is possibly mergeable now but the upgrade script might take a bit more

Before
----------------------------------------
_What is the old user-interface or technical-contract (as appropriate)?_
_For optimal clarity, include a concrete example such as a screenshot, GIF ([LICEcap](http://www.cockos.com/licecap/), [SilentCast](https://github.com/colinkeenan/silentcast)), or code-snippet._

After
----------------------------------------
_What changed? What is new old user-interface or technical-contract?_
_For optimal clarity, include a concrete example such as a screenshot, GIF ([LICEcap](http://www.cockos.com/licecap/), [SilentCast](https://github.com/colinkeenan/silentcast)), or code-snippet._

Technical Details
----------------------------------------
_If the PR involves technical details/changes/considerations which would not be manifest to a casual developer skimming the above sections, please describe the details here._

Comments
----------------------------------------
_Anything else you would like the reviewer to note_

### CODE REVIEW

File: CRM/Upgrade/Incremental/php/FiveSeventySix.php [hunk 1]
```diff
@@ -32,6 +34,69 @@ public function upgrade_5_76_alpha1($rev): void {
     $this->addTask('Add start_date to civicrm_mailing table', 'addColumn', 'civicrm_mailing', 'start_date', "timestamp NULL DEFAULT NULL COMMENT 'date on which this mailing was started.'");
     $this->addTask('Add end_date to civicrm_mailing table', 'addColumn', 'civicrm_mailing', 'end_date', "timestamp NULL DEFAULT NULL COMMENT 'date on which this mailing was completed.'");
     $this->addTask('Add status to civicrm_mailing table', 'addColumn', 'civicrm_mailing', 'status', "varchar(12) DEFAULT NULL COMMENT 'The status of this Mailing'");
+    $this->addTask('Update civicrm_mailing to permit deleting records from civicrm_mailing_job', 'updateNewCiviMailFields');
+  }
+
+  public static function updateNewCiviMailFields(CRM_Queue_TaskContext $ctx): bool {
+    [$minId, $maxId] = CRM_Core_DAO::executeQuery("SELECT coalesce(min(id),0), coalesce(max(id),0)
+      FROM civicrm_mailing ")->getDatabaseResult()->fetchRow();
+    for ($startId = $minId; $startId <= $maxId; $startId += self::MAILING_BATCH_SIZE) {
```
Comment (seamuslee001) [inline CRM/Upgrade/Incremental/php/FiveSeventySix.php:43]: @eileenmcnaughton I think for the upgrade tests we need to handle where $minId and $maxId == 0

Comment (eileenmcnaughton) [reply CRM/Upgrade/Incremental/php/FiveSeventySix.php:43]: ah thanks

**Comment by eileenmcnaughton** [comment]
@stesi561 - are you still up for taking a look at this

**Comment by stesi561** [comment]
Sorry too busy today but @jitendrapurohit can probably look at when he's online later today.

**Comment by eileenmcnaughton** [comment]
thanks @stesi561 @jitendrapurohit

**Comment by jitendrapurohit** [comment]
@eileenmcnaughton Have tested this on a site with ~3K rows in mailing table and 13K rows in mailing_job. The upgrade ran fine and dont see any issues after the upgrade.

Thanks.

**Review by jitendrapurohit — APPROVED** [review]

**Comment by eileenmcnaughton** [comment]
thanks @jitendrapurohit  - merging based on @jitendrapurohit's review

**Pull request merged**